lib/client.d.ts

```diff
@@ -0,0 +1,87 @@
+/// <reference types="node" />
+/// <reference types="Long" />
+
+import { EventEmitter } from 'events';
+import * as core from './dist';
+
+export type ClientOptions = {
+  connection: {
+    name?: string;
+    endPoint: string | core.TcpEndPoint | core.GossipSeed[];
+    settings: core.ConnectionSettings;
+  },
+  credentials: core.UserCredentials
+}
+
+export type EventPayload = {
+  eventId: string,
+  eventType: string,
+  metadata: any,
+  data: any,
+}
+
+export type DecodedEvent = {
+  stream: string,
+  eventId: string,
+  eventType: string,
+  metadata: any,
+  data: any,
+  originalStream: string,
+  originalPosition: core.Position,
+  originalEventId: string,
+  originalEventNumber: Long | number
+}
+
+export interface ActableEvent {
+  ack(events: core.ResolvedEvent|core.ResolvedEvent[]): void;
+  nack(events: core.ResolvedEvent|core.ResolvedEvent[], action: core.PersistentSubscriptionNakEventAction, reason: string): void;
```

Comment:
shouldn't these two omit the first argument (`events`)? Those methods are assigned to the event object, so why would they need events to be passed as argument 🤔